### PR TITLE
build(webpack): Change `IS_CI` to enable when in CI and snapshots

### LINKED ISSUE #21409: Self-hosted Sentry reports incorrect version (Acceptance Test.test)
## Important Details

How are you running Sentry?

On-Premise w/ Docker, version 20.10.0

## Description

Sentry displays `Acceptance Test test` on the bottom left corner as its version.

## Steps to Reproduce

Run `./install.sh` from the 20.10.0 tag on getsentry/onpremise repo and run `docker-compose up -d`. After logging in you'll see the incorrect version at the bottom left corner.

### What you expected to happen

It should show `20.10.0`

### Possible Solution

This is because us starting to set `CI=1` in Google Cloud Build during the wheel builds. Our JS code (specifically, `getDynamicTest`) uses this env variable incorrectly to detect pull request or acceptance test builds and during the compilation, this value gets hard-coded into the wheel.

As a stop-gap solution, I'll unset the `CI` variable during wheel builds but we should probably switch to using `VISUAL_SNAPSHOT_ENABLE` for detecting acceptance test snapshots.

First reported over at https://github.com/getsentry/onpremise/issues/696#issuecomment-709883317

### PR DESCRIPTION
**Pull request by billyvg**
This changes `IS_CI` to be enabled only when we are in a CI environment and we are using visual snapshots. We also rename the `IS_CI` constant in our frontend app to `IS_ACCEPTANCE_TEST` to better reflect this change (in the future we may want to add the env var to our `Makefile`). The reason for this change is because we are now starting to use CI for releases and this will cause our JS bundle to be in this test mode.

Fixes #21409

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -25,7 +25,10 @@ const {env} = process;
 const IS_PRODUCTION = env.NODE_ENV === 'production';
 const IS_TEST = env.NODE_ENV === 'test' || env.TEST_SUITE;
 const IS_STORYBOOK = env.STORYBOOK_BUILD === '1';
-const IS_CI = !!env.CI || !!env.TRAVIS;
+// This is used to stop rendering dynamic content for tests/snapshots
+// We want it in the case where we are running tests and it is in CI,
+// this should not happen in local
+const IS_CI = (!!env.CI || !!env.TRAVIS) && !!env.VISUAL_SNAPSHOT_ENABLE;
```
Comment (BYK) [inline webpack.config.js:31]: Do we really need the `!!` thing?

Comment (billyvg) [reply webpack.config.js:31]: No, we're using it to cast to a boolean, I'd prefer to keep it

Comment (BYK) [reply webpack.config.js:31]: How about this?

```suggestion
const IS_CI = Boolean((env.CI || env.TRAVIS) && env.VISUAL_SNAPSHOT_ENABLE);
```

Comment (billyvg) [reply webpack.config.js:31]: We're already using `!!` everywhere else though, and not that perf matters too much, but [`!!` performs marginally better than `Boolean()`: ](https://perf.link/#eyJpZCI6IjM5NGExbjFjemY4IiwidGl0bGUiOiJDYXN0aW5nIHRvIGJvb2xlYW4iLCJiZWZvcmUiOiJjb25zdCBkYXRhID0gWy4uLkFycmF5KDEwMDApLmtleXMoKV0iLCJ0ZXN0cyI6W3sibmFtZSI6IiEhIiwiY29kZSI6ImZvciAobGV0IGkgaW4gZGF0YSkge1xuICAhIWZhbHNlXG59IiwicnVucyI6WzE4MDAwLDEyMDAwLDEzMDAwLDgwMDAsMTAwMCwxMDAwLDIwMDAsNjAwMCwxMTAwMCwxMjAwMCw2MDAwLDkwMDAsMTMwMDAsMTAwMCwyMDAwLDYwMDAsOTAwMCwxMzAwMCwzMDAwLDYwMDAsMTAwMCwxMjAwMCwxMjAwMCwxMDAwLDE0MDAwLDYwMDAsMjAwMCwxMDAwLDEwMDAsMzAwMCwxODAwMCwxMDAwLDE3MDAwLDE2MDAwLDEyMDAwLDcwMDAsNTAwMCwxOTAwMCwxNDAwMCwzMDAwLDQwMDAsMTAwMCwxMDAwLDUwMDAsNTAwMCw4MDAwLDIwMDAsMTEwMDAsMTYwMDAsMTAwMCw4MDAwLDUwMDAsMTAwMDAsMTAwMCw1MDAwLDUwMDAsNjAwMCwxNTAwMCw5MDAwLDQwMDAsMTkwMDAsMTEwMDAsMTgwMDAsMTAwMCwxMDAwLDEzMDAwLDcwMDAsODAwMCw4MDAwLDE3MDAwLDEwMDAwLDE2MDAwLDE5MDAwLDMwMDAsMTcwMDAsMTYwMDAsMTgwMDAsNTAwMCwzMDAwLDE4MDAwLDE3MDAwLDE4MDAwLDE0MDAwLDMwMDAsMTAwMCw1MDAwLDEwMDAsNzAwMCwxNjAwMCw1MDAwLDEzMDAwLDYwMDAsMTkwMDAsMTkwMDAsMTAwMCw2MDAwLDgwMDAsMTMwMDAsNDAwMCw0MDAwXSwib3BzIjo4NDcwfSx7Im5hbWUiOiJCb29sZWFuIiwiY29kZSI6ImZvciAobGV0IGkgaW4gZGF0YSkge1xuICBCb29sZWFuKGZhbHNlKVxufSIsInJ1bnMiOlsxMDAwLDIwMDAsOTAwMCwxMDAwLDExMDAwLDQwMDAsMTIwMDAsMTAwMCwxMDAwMCw4MDAwLDEwMDAsMTAwMCw2MDAwLDQwMDAsMjAwMCwxMDAwLDMwMDAsMTIwMDAsMTAwMCwxMDAwLDgwMDAsNTAwMCwxMDAwLDEwMDAwLDEzMDAwLDEyMDAwLDUwMDAsMTAwMDAsMTMwMDAsNTAwMCwxMzAwMCwxMDAwLDEwMDAsMTMwMDAsODAwMCwxMzAwMCwxMDAwLDkwMDAsNTAwMCwyMDAwLDgwMDAsMTAwMCw0MDAwLDEwMDAsMzAwMCw0MDAwLDEwMDAsOTAwMCwxMzAwMCwxMDAwMCw1MDAwLDIwMDAsNjAwMCwxMzAwMCwxMDAwLDYwMDAsMjAwMCwxMTAwMCw3MDAwLDEwMDAwLDEwMDAsMjAwMCwxMTAwMCwxMTAwMCwxMDAwLDEzMDAwLDUwMDAsNDAwMCw2MDAwLDcwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwyMDAwLDExMDAwLDEzMDAwLDcwMDAsMTAwMCw3MDAwLDEyMDAwLDE0MDAwLDExMDAwLDEwMDAsNTAwMCwzMDAwLDEwMDAsMjAwMCw3MDAwLDEwMDAsMTAwMDAsMzAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwMCwzMDAwLDEwMDAsMTAwMCwyMDAwXSwib3BzIjo1NTIwfV0sInVwZGF0ZWQiOiIyMDIwLTEwLTE2VDIwOjAwOjU1Ljc0NVoifQ%3D%3D)

**Comment by billyvg** [comment]
#test-getsentry

**Comment by billyvg** [comment]
Not sure why `getsentry/javascript` fails w/ webpack, but `getsentry/acceptance` passes fine.

**Review by priscilawebdev — APPROVED** [review]

File: webpack.config.js (hunk 1 above)
Comment (BYK) [inline webpack.config.js:31]: Btw, I'd still rename `IS_CI` to be more precise like `IS_TESTING` or `IS_ACCEPTANCE_TESTS` or event `USE_FIXED_TEST_STRINGS`.

**Review by BYK — APPROVED** [review]
Nice, thanks!

Do you think it is safe to set `CI=1` now when building static assets? Asking because I see that you left the definition of it so I wonder if anything else is relying on this for UI behavior.

**Comment by billyvg** [comment]
Yeah, it's safe, I left it to be more explicit about the `IS_ACCEPTANCE_TEST` condition

**Pull request merged**